definitions/npm/firebase_v4.x.x/flow_v0.46.x-/firebase_v4.x.x.js

```diff
@@ -1,343 +1,402 @@
+declare class FacebookAuthProvider extends AuthProvider {
+  PROVIDER_ID: string,
+  credential(token: string): AuthCredential,
+  addScope(scope: string): FacebookAuthProvider,
+  setCustomParameters(customOAuthParameters: Object): FacebookAuthProvider
 }
 
 declare class GithubAuthProvider extends AuthProvider {
-  PROVIDER_ID: string;
-  credential(token: string): AuthCredential;
-  addScope(scope: string): GithubAuthProvider;
-  setCustomParameters(customOAuthParameters: Object): GithubAuthProvider;
+  PROVIDER_ID: string,
+  credential(token: string): AuthCredential,
+  addScope(scope: string): GithubAuthProvider,
+  setCustomParameters(customOAuthParameters: Object): GithubAuthProvider
 }
 
 declare class GoogleAuthProvider extends AuthProvider {
-  PROVIDER_ID: string;
-  credential(idToken?: string, accessToken?: string): AuthCredential;
-  addScope(scope: string): GoogleAuthProvider;
-  setCustomParameters(customOAuthParameters: Object): GoogleAuthProvider;
+  PROVIDER_ID: string,
+  credential(idToken?: string, accessToken?: string): AuthCredential,
+  addScope(scope: string): GoogleAuthProvider,
+  setCustomParameters(customOAuthParameters: Object): GoogleAuthProvider
 }
 
 declare class PhoneAuthProvider extends AuthProvider {
-  PROVIDER_ID: string;
-  constructor(auth?: Auth): PhoneAuthProvider;
-  credential(verificationId: string, verificationCode: string): AuthCredential;
-  verifyPhoneNumber(phoneNumber: string, applicationVerifier: ApplicationVerifier): Promise<string>;
+  PROVIDER_ID: string,
+  constructor(auth?: Auth): PhoneAuthProvider,
+  credential(verificationId: string, verificationCode: string): AuthCredential,
+  verifyPhoneNumber(
+    phoneNumber: string,
+    applicationVerifier: ApplicationVerifier
+  ): Promise<string>
 }
 
 declare class TwitterAuthProvider extends AuthProvider {
-  PROVIDER_ID: string;
-  credential(token: string, secret: string): AuthCredential;
-  setCustomParameters(customOAuthParameters: Object): this;
+  PROVIDER_ID: string,
+  credential(token: string, secret: string): AuthCredential,
+  setCustomParameters(customOAuthParameters: Object): this
 }
 
-declare type OAuthProvider = | FacebookAuthProvider
-                             | GithubAuthProvider
-                             | GoogleAuthProvider
-                             | TwitterAuthProvider;
-
-/****** database ******/
-declare type FirebaseValue = any;
-declare type OnCompleteCallback = (error: ?Object) => void;
-declare type QueryEventType = | 'value'
-                              | 'child_added'
-                              | 'child_changed'
-                              | 'child_removed'
-                              | 'child_moved';
-declare type Priority = string | number | null;
+declare type OAuthProvider =
+  | FacebookAuthProvider
+  | GithubAuthProvider
+  | GoogleAuthProvider
+  | TwitterAuthProvider
+
+/** **** database ******/
+declare type FirebaseValue =
```

Comment:
In my opinion I still think this should be type `any`. Even though you mentioned it can't be certain values. I think the developer should know what value is coming out of their .val() and they should cast it to the right type making sure that the correct value goes in or out. I think this is why in typescript for this is `any`. Having FirebaseValue not be `any` makes it hard to call update to with a value that has a type that I declared. I could be wrong though..